### PR TITLE
OCLOMRS-52: A user should be able to view paginated concepts in a particular source

### PR DESCRIPTION
**Pull request by shakirandagire**
# JIRA TICKET NAME:
[A user should be able to view paginated concepts in a particular source	
](https://issues.openmrs.org/browse/OCLOMRS-52)

# Summary:
As a user, I should be able to view paginated concepts in a particular source

### CODE REVIEW

File: src/components/dashboard/container/SpecificConcept.jsx [hunk 1]
```diff
@@ -15,15 +18,17 @@ export class SpecificConcept extends Component {
...
   constructor(props) {
     super(props);
 
     this.state = {
       searchInput: '',
+      offset: 2,
     };
-    this.onSearch = this.onSearch.bind(this);
-    this.onSubmit = this.onSubmit.bind(this);
+    autoBind(this);
```
Comment (mwaz) [inline src/components/dashboard/container/SpecificConcept.jsx]: I still think there is a better way of doing autoBind('functionOne', 'functionTwo')

File: src/tests/Dashboard/reducer/conceptReducer.test.js [hunk 2]
```diff
@@ -36,4 +39,29 @@ describe('Test suite for concepts reducer', () => {
       loading: false,
     });
   });
+
+  it('should handle SEARCH_CONCEPTS', () => {
+    expect(reducer(
```
Comment (Efosaok) [inline src/tests/Dashboard/reducer/conceptReducer.test.js]: You would want to make your it block messages more descriptive when writing tests for reducers, actions and components. Tests are usually the first place where someone onboarding into a codebase can look into and get an idea of what each unit function is doing, and you'll be helping out if your test case message is the actual description of what that function or unit block of code being tested is actually doing..

a better way you could write this could be

```
it sets redux state key loading and hasMore to false
on dispatching action type SEARCH_CONCEPTS.`
```
overall, I think you have done a great Job.

Comment (dkayiwa) [reply src/tests/Dashboard/reducer/conceptReducer.test.js]: I like it so much when the new guys on the team also start reviewing code! 👍

Comment (shakirandagire) [reply src/tests/Dashboard/reducer/conceptReducer.test.js]: Thank you for the feedback @Efosaok. I have made the relevant changes to my tests to make them more descriptive as suggested.
@dkayiwa we like this too. Thanks @Efosaok for commenting

File: src/components/dashboard/container/SpecificConcept.jsx (hunk 1 above)
Comment (shakirandagire) [reply src/components/dashboard/container/SpecificConcept.jsx]: @mwaz I chose to use autoBind(this) to help bind the onSearch and onSubmit methods to this instance. I had not navigated the way suggested to do the binding but will explore it in future

Comment (mwaz) [reply src/components/dashboard/container/SpecificConcept.jsx]: awesome..

**Comment by shakirandagire** [comment]
@dkayiwa I have fixed the merge conflicts

File: .travis.yml [hunk 3]
```diff
@@ -1,9 +1,11 @@
 language: node_js
 node_js:
   - "8.9.4"
-script:
-  - npm install
+install:
```
Comment (dkayiwa) [inline .travis.yml]: Are these changes required as part of this ticket?

Comment (shakirandagire) [reply .travis.yml]: I had issues with travis build terminating without running tests.
I researched and the solution was to restructure the travis.yml file.

Comment (dkayiwa) [reply .travis.yml]: That requires a separate ticket and pull request. Do you still remember the article i shared with you over the weekend? 😊
